symforce/opt/factor.h

```diff
@@ -124,10 +116,9 @@ class Factor {
    *                     be a subset of keys_to_func.
    */
   Factor(DenseHessianFunc hessian_func, const std::vector<Key>& keys_to_func,
-         const std::vector<Key>& keys_to_optimize);
-  Factor(SparseHessianFunc hessian_func, const std::vector<Key>& keys_to_func);
+         const std::vector<Key>& keys_to_optimize = {});
   Factor(SparseHessianFunc hessian_func, const std::vector<Key>& keys_to_func,
-         const std::vector<Key>& keys_to_optimize);
+         const std::vector<Key>& keys_to_optimize = {});
```

Comment:
I still think this is a marginally less clear API for a pretty marginal gain in simplicity - if we're going to do this though we should at least note in the doc comment for `keys_to_optimize` that passing the empty vector (the default) means to assume it equals `keys_to_func`.

Idk if @hayk-skydio has thoughts on the API here

Reply:
I would want to deprecate functions that take both `keys` and `keys_to_opt` later on in favor of the new `FactorKeys`.
This is just a first step toward that goal which I plan to break up into several PRs.
No need to approve this for now if this change alone is less clear, but hopefully it would make sense once the new class is in place.

Reply:
I think maybe it makes sense to just wait?  You can also make the FactorKeys change a multi-commit PR if you want, or make it a chain, either way

Reply:
Yeah, I'm ok with waiting. This PR was meant to just start a discussion and also let the CI do some checking for me.